pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -537,8 +551,8 @@ func (lbaas *LbaasV2) UpdateLoadBalancer(clusterName string, service *api.Servic
 
 	// Compose Set of member (addresses) that _should_ exist
 	addrs := map[string]empty{}
-	for _, nodeName := range nodeNames {
-		addr, err := getAddressByName(lbaas.compute, types.NodeName(nodeName))
+	for _, node := range nodes {
+		addr, err := nodeAddressForLB(node)
```

Comment:
This pr replaces the function that checks for unschedulable (hostsFromNodeList) with one that doesn't, I think. Servicecontroller is responsible for understand node readiness/schedulability, which it currently does at listing time via  `getNodeConditionPredicate`. Can you confirm that we aren't regressing anything? I don't want to push the burden of schedulable/unscheduleable/taints/readieness etc onto each cloudprovider module.